### DISCUSSION TITLE
[Beacon] Converting help buffer text into lists

### DISCUSSION THREAD

**Discussion by trev-dev** [meow-edit/meow#289] [Q&A] [answered]
I was able to wrap my head around this problem using multiple cursors, but not beacon mode. I am curious, how would you solve this problem using Meow?

The Sample:
```
‘trailing-whitespace’
     The face for highlighting excess spaces and tabs at the end of a
     line when ‘show-trailing-whitespace’ is non-‘nil’ (*note Useless
     Whitespace::).
‘escape-glyph’
     The face for displaying control characters and escape sequences
     (*note Text Display::).
‘homoglyph’
     The face for displaying lookalike characters, i.e., characters that
     look like but are not the characters being represented (*note Text
     Display::).
‘nobreak-space’
     The face for displaying no-break space characters (*note Text
     Display::).
‘nobreak-hyphen’
     The face for displaying no-break hyphen characters (*note Text
     Display::).
‘mode-line’
     This face is used for the mode line of the currently selected
     window, and for menu bars when toolkit menus are not used.  By
     default, it’s drawn with shadows for a raised effect on graphical
     displays, and drawn as the inverse of the default face on
     non-windowed terminals.
‘mode-line-inactive’
     Like ‘mode-line’, but used for mode lines of the windows other than
     the selected one (if ‘mode-line-in-non-selected-windows’ is
     non-‘nil’).  This face inherits from ‘mode-line’, so changes in
     that face affect mode lines in all windows.
‘mode-line-highlight’
     Like ‘highlight’, but used for mouse-sensitive portions of text on
     mode lines.  Such portions of text typically pop up tooltips (*note
     Tooltips::) when the mouse pointer hovers above them.
‘mode-line-buffer-id’
     This face is used for buffer identification parts in the mode line.
‘header-line’
     Similar to ‘mode-line’ for a window’s header line, which appears at
     the top of a window just as the mode line appears at the bottom.
     Most windows do not have a header line—only some special modes,
     such Info mode, create one.
‘header-line-highlight’
     Similar to ‘highlight’ and ‘mode-line-highlight’, but used for
     mouse-sensitive portions of text on header lines.  This is a
     separate face because the ‘header-line’ face might be customized in
     a way that does not interact well with ‘highlight’.
```

The desired outcome:
```lisp
`(trailing-whitespace)
`(escape-glyph)
`(homoglyph)
`(nobreak-space)
`(nobreak-hyphen)
`(mode-line)
`(mode-line-inactive)
`(mode-line-highlight)
`(mode-line-buffer-id)
`(header-line)
`(header-line-highlight)
```

I feel like this was challenging to do this elegantly with beacon for a couple of reasons:

1. Lack of simple, unique characters or words to find or select word/symbol for
2. Variance in the number of lines for each element
3. Inability to select a chunk of white-space. I solved this using `multiple-cursors` by:
    - Selecting a region
    - Using `mc/mark-all-in-region` for 5 spaces
    - Deleting the description text and newlines.
    - Leaving mc mode, re-selecting what's left. Use mc/edit lines to get a uniform cursor per line
    - The rest is trivial

**Comment by eshrh** [comment]
Lol at some point i bite the bullet and drop back to some good old regex `^'(.*)'` and elisp string concatenation...

**Comment by trev-dev** [comment]
I managed to do it with beacon by "scanning" forward and watching the beacon positions. If there's a better way I am open to it :)

![solution](https://user-images.githubusercontent.com/28788713/180078473-766aa7f4-9b92-436e-8840-ae8a81ed6db5.gif)

Cc: @eshrh

**Reply by eshrh** [reply]
I've got no idea, I was never particularly good at using the beacon anyway. Like I said my first instinct when i saw your post would have been to drop into my scratch buffer and get hacking haha

**Comment by fbergroth** [comment, marked as answer]
`M-x flush-lines` and input `^ ` (caret space) is another way to get rid of the text. After that a beacon follows nicely.

You can create a cursor on every symbol by doing (when beacon is active) `C-M-s ^‘ RET` and press `n` (or `meow-search`) to create the cursors.

**Reply by fbergroth** [reply]
> 3\. Inability to select a chunk of white-space. I solved this using `multiple-cursors` by:

(So it's possible to create cursors in the same way you did with multiple cursors by doing `C-M-s` to search for whitespaces (five spaces or leading ws) followed by `meow-search`.)

**Reply by trev-dev** [reply]
Both of these tips are amazing and simple to follow, nice!

**Comment by DogLooksGood** [comment]
https://user-images.githubusercontent.com/11796018/180100038-25d02a0b-d9b3-4e4d-8515-a4a5548bc64a.mp4

With another buffer, it's easier. Let me explain.

**Reply by DogLooksGood** [reply]
First, do a regexp-isearch. `meow-visit` and `meow-search` use the same ring with `C-M-s`. So if you search something like `^‘`, then you can mark all of them in BEACON state with `meow-search`.

Then you `F3` to start, copy the symbol to another buffer, go back, and `F4`.

**Reply by DogLooksGood** [reply]
Just noticed the order is wrong. So after yank, you should insert a newline at the beginning of line, instead of at the end of line. It could be a bug, I don't know why the order is reversed.

**Reply by DogLooksGood** [reply]
Just committed the patch to fix this. 0c977962f45d22fe8079e94ec7947f3b37663826

**Reply by trev-dev** [reply]
> something like ^‘

This was my first instinct until I realized that ‘ is not a character on my keyboard

Thanks for the solution!
